packages/grid/x-data-grid-pro/src/components/headerFiltering/constants.ts

```diff
@@ -0,0 +1,53 @@
+export const OPERATOR_LABEL_MAPPING: { [key: string]: string } = {
```

Comment:
That's a good idea, I actually thought to extend the existing way of handling these labels, i.e. convert `value` for the operators (which are symbols) to strings (all in [gridNumericOperators.ts](https://github.com/mui/mui-x/blob/master/packages/grid/x-data-grid/src/colDef/gridNumericOperators.ts)), then we can define these literals in `localeTextConstants.ts` below these:
https://github.com/mui/mui-x/blob/0f70a86ba1816e9a3c2dc6383c275bc49d964fec/packages/grid/x-data-grid/src/constants/localeTextConstants.ts#L59-L71

And use them like this (as being done in GridFilterForm):
https://github.com/mui/mui-x/blob/0f70a86ba1816e9a3c2dc6383c275bc49d964fec/packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterForm.tsx#L517-L520

That way we can:
1. Centralize labels for both regular filter form and header filters
2. Allow all the labels to be translated

But since this seems a breaking change, I [decided](https://github.com/mui/mui-x/pull/7760#discussion_r1171326330) to postpone it till v7 and use constants for now.

Do you think this a viable approach or do you have more use-cases of `headerFilterLabel` in mind which are not possible with the approach I mentioned?

Reply:
Regarding numeric operators - I think there's nothing blocking us from adding them to `localeTextConstants`:
```diff
filterOperatorIsAnyOf: 'is any of',
+'filterOperator=': '=',
```
And then removing the `label` in `gridNumericOperators`. I don't see this as a breaking change, unless I'm missing something

For the header filter labels, we can use the same approach - add `headerLabel`, and then add `headerFilterOperator${operator}` to `localeTextConstants`.
The `headerLabel` is optional (same as `label`), and if it's not defined - we call:

```tsx
apiRef.current.getLocaleText(`headerFilterOperator${capitalize(item.operator!)}
```

Does it make sense?

Reply:
> Regarding numeric operators - I think there's nothing blocking us from adding them to localeTextConstants:

Correct, I just wanted them to look a bit cleaner, so instead of `filterOperator=` => `filterOperatorEquals` which would require updating `operator.value` for the respective operators.

I was also staying away from introducing new locale text constants for the operators whose values would be exactly the same as `filterOperator${operator}` (doesn't make a lot of sense to have duplicate values, unless they could become different for some use-cases).

But, I agree that doing so would allow more customization by possibly having different labels for header filters and normal filters and also to the developers who want to define/customize custom filter operators.

Reply:
Done ✅ in https://github.com/mui/mui-x/pull/7760/commits/fd2264418a0107bb2f5cc197661a7fbc1389b0c0 and https://github.com/mui/mui-x/pull/7760/commits/49086949274396cd42940a65e15bbb725eec5d37